# 01 — Data Collection
## AI Adoption in U.S. Small and Medium Enterprises (2019–2024)

---

### Project Overview

**Business Problem**  
Small and medium sized enterprises (SMEs) represent 99.9% of all U.S. businesses and employ nearly half the private sector workforce. Despite the rapid democratization of AI tools especially after the launch of ChatGPT in late 2022 AI adoption among SMEs remains uneven and poorly understood. This project investigates *who* is adopting AI, *when*, *why*, and *what impact* it has on productivity and revenue growth.

**Research Questions**
1. How has AI adoption evolved among U.S. SMEs between 2019 and 2024?
2. Which industries and states lead in adoption, and which lag behind?
3. What organizational factors (size, cloud readiness, digital maturity, CEO background) predict AI adoption?
4. Does AI adoption translate into measurable productivity and revenue gains?
5. What are the primary barriers preventing SMEs from adopting AI?

**Dataset Overview**  
The analysis uses three complementary datasets spanning 2019–2024:
- **Main dataset**: 5,000 company year observations across 50 U.S. states and 10+ industries
- **Time-series dataset**: Annual adoption and investment metrics broken down by industry (60 rows)
- **National dataset**: Year-level aggregate statistics for the full SME population (6 rows)

**Project Goals**
- Understand the macro trend of AI adoption in U.S. SMEs from 2019 to 2024
- Identify key predictors of AI adoption using statistical and machine learning methods
- Quantify the productivity and revenue impact of AI investment
- Produce actionable insights for SME owners, policymakers, and technology vendors

---
## 1. Data Sources

The dataset was constructed by combining and synthesizing data from the following authoritative sources:

| Source | Organization | Description | Coverage |
|--------|-------------|-------------|----------|
| **Annual Business Survey (ABS)** | U.S. Census Bureau | Provides business characteristics, innovation, and technology adoption metrics for U.S. employer firms. Includes breakdowns by industry, geography, and firm size. | National, 2017–present |
| **Global AI Survey** | McKinsey & Company | Annual survey of executives across industries and geographies on AI adoption, investment, and perceived business impact. SME-specific cuts extracted. | Global, 2017–present |
| **AI Index Report** | Stanford HAI | Comprehensive annual report tracking AI development, adoption, and economic impact across sectors and countries, including SME-specific analyses. | Global, 2018–present |
| **Occupational Employment Statistics** | U.S. Bureau of Labor Statistics (BLS) | Tracks employment by occupation and industry, used to derive technology workforce percentages and wage data for SMEs. | National, annual |
| **Small Business Economic Profiles** | U.S. Small Business Administration (SBA) | State- and industry-level profiles of small business activity, revenue, employment, and technology investment patterns. | National & state-level, annual |

---
## 2. Load Data

In [2]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

# Display settings
pd.set_option('display.max_columns', 30)
pd.set_option('display.float_format', '{:,.2f}'.format)
pd.set_option('display.width', 120)

print('Libraries loaded successfully.')

Libraries loaded successfully.


In [3]:
# ── Load all three datasets ──────────────────────────────────────────────────
df_main       = pd.read_csv('../data/raw/sme_ai_adoption_main.csv')
df_timeseries = pd.read_csv('../data/raw/sme_ai_adoption_timeseries.csv')
df_national   = pd.read_csv('../data/raw/sme_ai_adoption_national.csv')

print('Datasets loaded:')
print(f'  df_main       : {df_main.shape}')
print(f'  df_timeseries : {df_timeseries.shape}')
print(f'  df_national   : {df_national.shape}')

Datasets loaded:
  df_main       : (5000, 22)
  df_timeseries : (60, 6)
  df_national   : (6, 7)


### 2.1 Main Dataset — `sme_ai_adoption_main.csv`

In [4]:
# Shape and first rows
print(f'Shape: {df_main.shape}')
df_main.head(5)

Shape: (5000, 22)


,company_id,year,state,industry,size_category,employees,revenue_m_usd,tech_invest_pct,tech_invest_usd,tech_emp_pct,years_in_business,founded_year,cloud_adopted,has_data_strategy,ceo_tech_background,digital_maturity,ai_adopted,ai_tools_count,ai_investment_usd,productivity_gain_pct,revenue_growth_pct,primary_barrier
0,SME_00001,2022,North Carolina,Hospitality & Food,Micro (1-9),3,0.05,0.02,925.00,0.05,23,1999,1,0,0,2.95,0,0,0.00,0.00,8.96,Lack of expertise
1,SME_00002,2022,California,Agriculture,Upper-Mid (250-500),264,13.99,0.05,"736,499.00",0.16,32,1990,1,0,1,4.64,0,0,0.00,0.00,9.47,Data quality
2,SME_00003,2021,North Carolina,Education,Medium (50-249),84,1.32,0.06,"76,504.00",0.05,34,1987,1,1,1,5.34,0,0,0.00,0.00,-7.94,Security concerns
3,SME_00004,2024,Ohio,Finance & Insurance,Small (10-49),34,3.18,0.01,"27,835.00",0.13,20,2004,1,0,0,4.08,1,7,"14,164.00",21.07,21.96,NaN
4,SME_00005,2020,Iowa,Education,Micro (1-9),1,0.05,0.03,"1,288.00",0.13,14,2006,1,1,0,4.41,0,0,0.00,0.00,7.14,Lack of expertise


In [5]:
# Column data types
print('Data types:')
print(df_main.dtypes)

Data types:
company_id                   str
year                       int64
state                        str
industry                     str
size_category                str
employees                  int64
revenue_m_usd            float64
tech_invest_pct          float64
tech_invest_usd          float64
tech_emp_pct             float64
years_in_business          int64
founded_year               int64
cloud_adopted              int64
has_data_strategy          int64
ceo_tech_background        int64
digital_maturity         float64
ai_adopted                 int64
ai_tools_count             int64
ai_investment_usd        float64
productivity_gain_pct    float64
revenue_growth_pct       float64
primary_barrier              str
dtype: object


In [6]:
# Descriptive statistics
df_main.describe(include='all').T

,count,unique,top,freq,mean,std,min,25%,50%,75%,max
company_id,5000,5000,SME_00001,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN
year,"5,000.00",NaN,NaN,NaN,"2,022.00",1.63,"2,019.00","2,021.00","2,022.00","2,023.00","2,024.00"
state,5000,30,California,251,NaN,NaN,NaN,NaN,NaN,NaN,NaN
industry,5000,10,Agriculture,541,NaN,NaN,NaN,NaN,NaN,NaN,NaN
size_category,5000,4,Micro (1-9),2283,NaN,NaN,NaN,NaN,NaN,NaN,NaN
employees,"5,000.00",NaN,NaN,NaN,62.90,103.73,1.00,5.00,16.00,48.00,499.00
revenue_m_usd,"5,000.00",NaN,NaN,NaN,3.12,7.69,0.05,0.17,0.57,2.49,158.24
tech_invest_pct,"5,000.00",NaN,NaN,NaN,0.04,0.02,0.01,0.03,0.04,0.05,0.14
tech_invest_usd,"5,000.00",NaN,NaN,NaN,"162,469.47","463,513.14",250.00,"4,833.75","19,417.50","98,526.50","8,553,149.00"
tech_emp_pct,"5,000.00",NaN,NaN,NaN,0.11,0.06,0.00,0.07,0.10,0.14,0.41


### 2.2 Time-Series Dataset — `sme_ai_adoption_timeseries.csv`

In [7]:
print(f'Shape: {df_timeseries.shape}')
df_timeseries.head(5)

Shape: (60, 6)


,year,industry,adoption_rate,avg_ai_investment_usd,avg_productivity_gain_pct,n_companies
0,2019,Technology & Software,0.53,"58,808.00",30.06,57
1,2019,Finance & Insurance,0.48,"104,047.00",25.48,46
2,2019,Healthcare & Biotech,0.20,"40,285.00",24.70,45
3,2019,Retail & E-commerce,0.27,"310,271.00",22.09,48
4,2019,Manufacturing,0.35,"94,894.00",22.11,49


In [8]:
print('Data types:')
print(df_timeseries.dtypes)

Data types:
year                           int64
industry                         str
adoption_rate                float64
avg_ai_investment_usd        float64
avg_productivity_gain_pct    float64
n_companies                    int64
dtype: object


In [9]:
df_timeseries.describe().T

,count,mean,std,min,25%,50%,75%,max
year,60.00,"2,021.50",1.72,"2,019.00","2,020.00","2,021.50","2,023.00","2,024.00"
adoption_rate,60.00,0.41,0.24,0.04,0.19,0.37,0.55,0.97
avg_ai_investment_usd,60.00,"100,829.55","86,641.02","5,381.00","43,336.75","81,193.00","109,486.50","386,748.00"
avg_productivity_gain_pct,60.00,26.03,2.73,16.84,24.86,25.92,26.83,34.93
n_companies,60.00,83.33,25.87,45.00,57.75,80.50,108.00,129.00


### 2.3 National Dataset — `sme_ai_adoption_national.csv`

In [10]:
print(f'Shape: {df_national.shape}')
df_national.head(6)

Shape: (6, 7)


,year,adoption_rate,avg_ai_investment_usd,avg_productivity_gain_pct,avg_revenue_growth_pct,pct_cloud_adopted,n_companies
0,2019,0.27,"97,057.00",25.64,9.57,0.27,501
1,2020,0.30,"147,658.00",26.83,11.10,0.41,575
2,2021,0.37,"79,081.00",26.25,10.63,0.44,765
3,2022,0.44,"161,430.00",26.25,11.16,0.55,900
4,2023,0.50,"87,573.00",25.74,12.65,0.62,1111
5,2024,0.54,"112,047.00",25.82,12.34,0.69,1148


In [11]:
print('Data types:')
print(df_national.dtypes)

Data types:
year                           int64
adoption_rate                float64
avg_ai_investment_usd        float64
avg_productivity_gain_pct    float64
avg_revenue_growth_pct       float64
pct_cloud_adopted            float64
n_companies                    int64
dtype: object


In [12]:
df_national.describe().T

,count,mean,std,min,25%,50%,75%,max
year,6.00,"2,021.50",1.87,"2,019.00","2,020.25","2,021.50","2,022.75","2,024.00"
adoption_rate,6.00,0.40,0.11,0.27,0.32,0.41,0.49,0.54
avg_ai_investment_usd,6.00,"114,141.00","33,439.34","79,081.00","89,944.00","104,552.00","138,755.25","161,430.00"
avg_productivity_gain_pct,6.00,26.09,0.45,25.64,25.76,26.04,26.25,26.83
avg_revenue_growth_pct,6.00,11.24,1.13,9.57,10.75,11.13,12.04,12.65
pct_cloud_adopted,6.00,0.50,0.15,0.27,0.42,0.50,0.60,0.69
n_companies,6.00,833.33,269.22,501.00,622.50,832.50,"1,058.25","1,148.00"


---
## 3. Data Sources Summary

In [13]:
# ── Summary statistics per year ──────────────────────────────────────────────
summary_by_year = (
    df_main
    .groupby('year')
    .agg(
        n_companies       = ('company_id', 'count'),
        adoption_rate     = ('ai_adopted', 'mean'),
        avg_revenue_m_usd = ('revenue_m_usd', 'mean'),
        avg_employees     = ('employees', 'mean'),
        avg_ai_invest_usd = ('ai_investment_usd', 'mean'),
        pct_cloud         = ('cloud_adopted', 'mean'),
        pct_data_strategy = ('has_data_strategy', 'mean'),
    )
    .reset_index()
)

# Format for display
summary_by_year['adoption_rate']     = (summary_by_year['adoption_rate'] * 100).round(1).astype(str) + '%'
summary_by_year['pct_cloud']         = (summary_by_year['pct_cloud'] * 100).round(1).astype(str) + '%'
summary_by_year['pct_data_strategy'] = (summary_by_year['pct_data_strategy'] * 100).round(1).astype(str) + '%'

print('Year-by-Year Summary:')
summary_by_year

Year-by-Year Summary:


,year,n_companies,adoption_rate,avg_revenue_m_usd,avg_employees,avg_ai_invest_usd,pct_cloud,pct_data_strategy
0,2019,501,26.5%,2.99,61.14,"25,765.59",26.9%,23.4%
1,2020,575,30.1%,3.91,69.82,"44,425.78",40.9%,30.1%
2,2021,765,36.9%,2.78,59.13,"29,151.38",44.2%,32.9%
3,2022,900,44.2%,3.50,63.52,"71,387.80",55.4%,39.4%
4,2023,1111,50.1%,2.66,58.00,"43,904.75",61.7%,43.1%
5,2024,1148,53.8%,3.18,66.96,"60,317.97",69.2%,49.7%


---
## 4. Key Stats Callout

In [14]:
# ── Key dataset statistics ───────────────────────────────────────────────────
total_records    = len(df_main)
years_covered    = sorted(df_main['year'].unique())
n_industries     = df_main['industry'].nunique()
n_states         = df_main['state'].nunique()
overall_adoption = df_main['ai_adopted'].mean() * 100

print('=' * 55)
print('  KEY DATASET STATISTICS')
print('=' * 55)
print(f'  Total Records         : {total_records:,}')
print(f'  Years Covered         : {years_covered[0]} – {years_covered[-1]}')
print(f'  Number of Years       : {len(years_covered)}')
print(f'  Industries            : {n_industries}')
print(f'  States Represented    : {n_states}')
print(f'  Overall Adoption Rate : {overall_adoption:.1f}%')
print(f'  Adopters (count)      : {df_main["ai_adopted"].sum():,}')
print(f'  Non-Adopters (count)  : {(df_main["ai_adopted"] == 0).sum():,}')
print('=' * 55)

  KEY DATASET STATISTICS
  Total Records         : 5,000
  Years Covered         : 2019 – 2024
  Number of Years       : 6
  Industries            : 10
  States Represented    : 30
  Overall Adoption Rate : 43.2%
  Adopters (count)      : 2,161
  Non-Adopters (count)  : 2,839


---
## 5. Variable Dictionary

Complete description of all columns in the main dataset (`sme_ai_adoption_main.csv`):

| Column | Type | Description |
|--------|------|-------------|
| `company_id` | string | Unique anonymized company identifier (e.g., SME_00001) |
| `year` | int | Survey year (2019–2024) |
| `state` | string | U.S. state where the company is headquartered |
| `industry` | string | Industry classification (e.g., Technology & Software, Retail & E-commerce) |
| `size_category` | string | Company size tier: Micro (1–9), Small (10–49), Medium (50–249), Upper-Mid (250–500) |
| `employees` | int | Number of full-time employees at time of survey |
| `revenue_m_usd` | float | Annual revenue in millions of USD |
| `tech_invest_pct` | float | Technology investment as a percentage of total revenue (0–1 scale) |
| `tech_invest_usd` | float | Technology investment in USD |
| `tech_emp_pct` | float | Percentage of employees in technology-related roles (0–1 scale) |
| `years_in_business` | int | Number of years the company has been operating |
| `founded_year` | int | Year the company was founded |
| `cloud_adopted` | int | Binary flag: 1 if company uses cloud services, 0 otherwise |
| `has_data_strategy` | int | Binary flag: 1 if company has a formal data strategy, 0 otherwise |
| `ceo_tech_background` | int | Binary flag: 1 if CEO/founder has a technology background, 0 otherwise |
| `digital_maturity` | float | Digital maturity score on a 1–10 scale (higher = more mature) |
| `ai_adopted` | int | Binary target variable: 1 if company has adopted AI tools, 0 otherwise |
| `ai_tools_count` | int | Number of distinct AI tools/applications in use (0 if not adopted) |
| `ai_investment_usd` | float | Annual AI-specific investment in USD (0 if not adopted) |
| `productivity_gain_pct` | float | Self-reported productivity gain attributed to technology/AI (%) |
| `revenue_growth_pct` | float | Year-over-year revenue growth rate (%) |
| `primary_barrier` | string | Main barrier preventing AI adoption (e.g., Cost, Lack of expertise, Data quality) |